File: problems-extend-adapt/2401-2500/2470_count-fixed-lcm-subarrays/solution.go
func countFixedLCMSubarrays(nums []int, k int) int {
	// Anchor the left endpoint and sweep right, carrying the running
	// lcm of nums[i..j]: it only ever grows (each new element can
	// raise it, never lower it). Once it exceeds k, every later lcm
	// in this sweep is larger still, so k is unreachable — break.
	// Each j where the lcm equals k is one counted subarray.
	gcd := func(a, b int) int {
		for b != 0 {
			a, b = b, a%b
		}
		return a
	}
	n := len(nums)
	total := 0
	for i := 0; i < n; i++ {
		l := 1
		for j := i; j < n; j++ {
			l = l / gcd(l, nums[j]) * nums[j]
			if l > k {
				break
			}
			if l == k {
				total++
			}
		}
	}
	return total
}

File: problems-extend-adapt/2401-2500/2470_count-fixed-lcm-subarrays/solution.rs
impl Solution {
    pub fn count_fixed_lcm_subarrays(nums: Vec<i32>, k: i32) -> i32 {
        // Anchor the left endpoint and sweep right, carrying the running
        // lcm of nums[i..j]: it only ever grows (each new element can
        // raise it, never lower it). Once it exceeds k, every later lcm
        // in this sweep is larger still, so k is unreachable — break.
        // Each j where the lcm equals k is one counted subarray.
        let gcd = |mut a: i32, mut b: i32| -> i32 {
            while b != 0 {
                let t = a % b;
                a = b;
                b = t;
            }
            a
        };
        let n = nums.len();
        let mut total = 0i64;
        for i in 0..n {
            let mut l = 1i32;
            for j in i..n {
                l = l / gcd(l, nums[j]) * nums[j];
                if l > k {
                    break;
                }
                if l == k {
                    total += 1;
                }
            }
        }
        total as i32
    }
}
